File: mainActivity/src/main/java/net/airvantage/model/alert/v1/AlertRule.java
package net.airvantage.model.alert.v1;

import net.airvantage.utils.Predicate;

import java.util.List;

public class AlertRule {
    public String uid;
    public boolean active = true;
    public String name;
    public String eventType;
    public List<Condition> conditions;

    public static Predicate<AlertRule> isNamed(final String name) {
        return new Predicate<AlertRule>() {
            @Override
            public boolean matches(AlertRule item) {
                return name.equals(item.name);
            }
        };
    }
}

File: mainActivity/src/main/java/net/airvantage/model/alert/v1/AlertRule.kt
package net.airvantage.model.alert.v1


class AlertRule {
    var uid: String? = null
    var active = true
    var name: String? = null
    var eventType: String? = null
    var conditions: MutableList<Condition>? = null



    companion object {

        fun isNamed(name: String): net.airvantage.utils.Predicate<AlertRule> {
            return  { item -> name == item.name }
        }
    }
}
